### ISSUE TITLE
Rufus, Windows To Go random file transfer error - Suggestion

### ISSUE THREAD

**Issue by Laboulett** [pbatard/rufus#1593]
﻿

Checklist
---------
- [X] I looked at https://github.com/pbatard/rufus/wiki/FAQ to see if my question has already been answered.
- [X] I performed a search in the issue tracker for similar issues using keywords relevant to my problem, such as the error message I got from the log.
- [X] I clicked the 'Log' button or pressed <kbd>Ctrl</kbd>-<kbd>L</kbd> in Rufus, and copy/pasted the log into the line that says `<FULL LOG>` below.
- [X] The log I am copying is the FULL log, starting with the line `Rufus version: x.y.z` - I have NOT removed any part of it.

Additionally (if applicable):
- [ ] I ran a bad blocks check, by clicking _Show advanced format options_ then _Check device for bad blocks_, and confirmed that my USB is not defective.
- [X] I also tried one or more of the following:
  - [X] Using a different USB drive.
  - [X] Plugging the USB into a different port.
  - [ ] Running Rufus on a different computer.
- [X] If using an image, I clicked on the `(✓)` button to compute the MD5, SHA1 and SHA256 checksums, which are therefore present in the log I copied. I confirmed, by performing an internet search, that these values match the ones from the official image.

Issue description
-----------------
I tried to install a Windows 10 Pro on a Micro SD card of 512GB using three different types of usb card readers using the "Windows To Go" option of Rufus.
This is not the first time I used Rufus to make a "Windows to Go" installation.
I already have an external HDD USB 3.0 and it working fine.
After some time, during the extraction process, I obtain a random "Apply error: ........." on a random file.
Then the error is not related to a specific file.
If I try multiple times, the error usually occurs on a different file.
Sometimes I can copy 500MB, sometimes 800MB, sometimes 600MB.
In any case, the error seems to be related to a file transfer bottleneck and this bottleneck occurs mainly when there are many files of few bytes.
Can you add a retry button instead of an automatic retry? because I think the error raised when the device is occupied.

Logs
----

* [rufus_errore.log](https://github.com/pbatard/rufus/files/5026965/rufus_errore.log)
* [rufus_errore_2.log](https://github.com/pbatard/rufus/files/5026999/rufus_errore_2.log)
* [rufus_errore_3.log](https://github.com/pbatard/rufus/files/5027276/rufus_errore_3.log)

**Comment by pbatard** [comment]
> Can you add a retry button instead of an automatic retry?

I'm afraid I can't. Because what you need to understand is that, when creating a Windows To Go drive, it's not Rufus that performs the individual file extraction from `install.wim`/`install.esd` but the Windows API (from `wimapi.dll`), to which we feed the whole `install.wim`/`install.esd`.

So the retry comes the the Windows WIM extraction API, which is a simple black box over which we have no control. All Rufus does it tell Windows _"Please apply the files from this image onto this destination drive"_, and we really have no control further than that, apart from getting the names of the files that Windows is applying or the error message that Windows produces in case the is a problem.

So, if I were to add a retry button in Rufus, it could only be for a whole retry of applying the image to the target drive, and not for an individual file (because there really is no API to tell Windows to continue applying an image starting with a specific file). As such, that retry would be pretty pointless.

In other words, if you want a better retry on WIM application error, you need to contact Microsoft support and ask them to improve the WIM API, as you would find you will get the same issue with `dism` (which uses the same API), and there's nothing I can do in Rufus to improve on the limitations of Microsoft's proprietary APIs.

As a result, I will close this issue.
